.github/workflows/tests.yml

```diff
@@ -387,32 +397,39 @@ jobs:
           node-version: ${{ matrix.node }}
       - name: Monorepo install
         uses: ./.github/actions/yarn-nm-install
+      - name: Monorepo build
+        uses: ./.github/actions/run-build
       - uses: ./.github/actions/run-api-tests
         with:
           dbOptions: '--dbclient=${{ matrix.db_client }} --dbhost=localhost --dbport=3306 --dbname=strapi_test --dbusername=strapi --dbpassword=strapi'
           runEE: true
+          jestOptions: '--shard=${{ matrix.shard }}'
 
   api_ee_sqlite:
     runs-on: ubuntu-latest
-    needs: [changes, lint, typescript, unit_back, unit_front]
-    name: '[EE] API Integration (sqlite, client: ${{ matrix.sqlite_pkg }}, node: ${{ matrix.node }})'
+    needs: [changes, build, typescript, unit_back, unit_front]
+    name: '[EE] API Integration (sqlite, client: ${{ matrix.sqlite_pkg }}, node: ${{ matrix.node }}, shard: ${{ matrix.shard }}'
     if: needs.changes.outputs.backend == 'true' && github.event.pull_request.head.repo.full_name == github.repository && !(github.actor == 'dependabot[bot]' || github.actor == 'dependabot-preview[bot]')
     env:
       STRAPI_LICENSE: ${{ secrets.strapiLicense }}
     strategy:
       matrix:
-        node: [16, 18, 20]
+        node: [18, 20]
         sqlite_pkg: ['better-sqlite3', 'sqlite3']
+        shard: [1/6, 2/6, 3/6, 4/6, 5/6, 6/6]
```

Comment:
We determined that the increase to using 2 shards will be fine, but I worry 6 could push us over our limit since we do use a significant amount already. Let's wait to see how the other PR goes for a few days before adding this.

Reply:
@innerdvations Because of the Caching the CI time is down